flutter_local_notifications/lib/src/platform_flutter_local_notifications.dart

```diff
@@ -542,35 +488,12 @@ class MacOSFlutterLocalNotificationsPlugin
 
   /// Schedules a notification to be shown at the specified date and time
   /// relative to a specific time zone.
-  Future<void> zonedSchedule(
-    int id,
-    String title,
-    String body,
-    TZDateTime scheduledDate,
-    MacOSNotificationDetails notificationDetails, {
-    String payload,
-    ScheduledNotificationRepeatFrequency scheduledNotificationRepeatFrequency,
-  }) async {
-    validateId(id);
-    validateDateIsInTheFuture(scheduledDate);
-    final Map<String, Object> serializedPlatformSpecifics =
-        notificationDetails?.toMap() ?? <String, Object>{};
-    await _channel.invokeMethod(
-        'zonedSchedule',
-        <String, Object>{
-          'id': id,
-          'title': title,
-          'body': body,
-          'platformSpecifics': serializedPlatformSpecifics,
-          'payload': payload ?? '',
-        }
-          ..addAll(scheduledDate.toMap())
-          ..addAll(scheduledNotificationRepeatFrequency == null
-              ? <String, Object>{}
-              : <String, Object>{
-                  'scheduledNotificationRepeatFrequency':
-                      scheduledNotificationRepeatFrequency.index
-                }));
+  Future<void> zonedSchedule(List<NotificationData> notifications) async {
```

Comment:
Concerned about the naming of `NotificationData` as it's very similar to `NotificationDetails`. Do you have other ideas for the name?

Reply:
yeah I wasn't completely happy about that one either. `Notification` is already taken in Flutter, so users of this lib would have to import with a prefix, which (I assumed) isn't desired. 

how about `LocalNotification`?

Reply:
Hmm I might have to think more about this. One option is to perhaps rename the existing `NotificationDetails`

Reply:
Unfortunately I can't think of a name and the more I think I about this and the PR, the more I wonder about if this should be PR should be part of the plugin.

The options for the name so far are generic that imply it could be used for other methods but it can't be.

The introduction of the class itself brings a bit inconsistency on how notification content is passed to be displayed. If a new method is introduced as part of the cross-platform facing API for displaying notifications, it should be something that can be added to the platform interface as well (note: `schedule` wasn't added as it was going to deprecated). However, this can't really be because the object it requires contains knowledge of other platforms. This may well be to do with how the reason this method exists right now is because it's a "workaround" for a performance issue using platform channels on Android. It would not otherwise have been part of the "core" APIs for the plugin.

With all the above in mind and unless you have ideas to address what I mentioned without breaking changes, I'm currently leaning towards leaving this out and including a link to this PR in the readme for those that would need this level of performance optimisation